sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java

```diff
@@ -756,12 +769,54 @@ synchronized void checkPendingLogs(@NonNull GroupState groupState) {
             return;
         }
         long pendingLogCount = groupState.mPendingLogCount;
-        AppCenterLog.debug(LOG_TAG, "checkPendingLogs(" + groupState.mName + ") pendingLogCount=" + pendingLogCount);
-        if (pendingLogCount >= groupState.mMaxLogsPerBatch) {
+        long batchTimeInterval = groupState.mBatchTimeInterval;
+        AppCenterLog.debug(LOG_TAG, String.format("checkPendingLogs(%s) pendingLogCount=%s batchTimeInterval=%s", groupState.mName, pendingLogCount, batchTimeInterval));
+
+        /* If the interval is custom. */
+        if (batchTimeInterval > MINIMUM_TRANSMISSION_INTERVAL) {
+            long now = System.currentTimeMillis();
+            long startTimer = SharedPreferencesManager.getLong(START_TIMER_PREFIX + groupState.mName);
+
+            /* The timer isn't started, so start it and store the current time. */
+            if (startTimer == 0) {
+                SharedPreferencesManager.putLong(START_TIMER_PREFIX + groupState.mName, now);
+                AppCenterLog.debug(LOG_TAG, "The timer value for " + groupState.mName + " has been saved.");
+            }
+
+            /* Handle invalid values (start time in the future). */
+            else if (startTimer > now) {
+                SharedPreferencesManager.remove(START_TIMER_PREFIX + groupState.mName);
+                AppCenterLog.debug(LOG_TAG, "Invalid timer value for " + groupState.mName + " channel has been removed.");
+            }
+
+            /* If the interval is over. */
+            else if (startTimer + batchTimeInterval < now) {
+
+                /* Send all logs without any additional timers. */
+                if (pendingLogCount > 0) {
+                    triggerIngestion(groupState);
+                } else {
+
+                    /* Remove the time only when there are no logs left. */
+                    SharedPreferencesManager.remove(START_TIMER_PREFIX + groupState.mName);
+                    AppCenterLog.debug(LOG_TAG, "The timer for " + groupState.mName + " channel finished.");
+                }
+                return;
+            }
+
+            /* We still have to wait for the rest of the interval. */
+            else {
+                batchTimeInterval -= now - startTimer;
+            }
+        } else if (pendingLogCount >= groupState.mMaxLogsPerBatch) {
```

Comment:
If we check pending logs after sending 1 of the 3 concurrent batches, and we still have more logs to sent after daily timer due. We still have to send the remaining logs immediately (if we have to send like 10 batches once a day). We might need a boolean for checkpendinglogs to allow this run or something like that.